### DISCUSSION TITLE
Using AI for searching issues

### DISCUSSION THREAD

**Discussion by niri81** [community/community#57548] [Projects and Issues]
### Select Topic Area

Product Feedback

### Select Feature Area

- [X] Issues (The feature that allows you to track ideas, feedback, tasks, or bugs for work on GitHub. Not for any issue you are having).
- [ ] Projects (For planning and tracking work on GitHub - not any open source or other type of project).

### Body

Hi all 👋🏻 ,
while searching an organizational repo with about 50 issues open, I managed to open an issue that was already there and written in slightly different terms.

When we search for issues before we open one, we already have a pretty good idea of what the title would be and we type in some of the keywords to find out if this issue has already been created. Based on our idea of how the issue will be structured and what we'll write in it, the issue we open may be basically a duplicate that we didn't find because, for example, we used a synonym in the search. Especially with an increasing number of open issues (+closed issues in the solution search), it may turn out that the current search tools are not sufficient to perform a time- and labor-saving search.

Therefore, I would like to put forward the following **feature request**: 
By using AI, the issues of the repo could be filtered and similar issues to what is being searched for could be presented to the user (both in search and issue creation, with a request for confirmation before the issue is created to reduce the likelihood of duplicates occurring). I think there are many more features that can be added, but this would be a very good start, especially for open source repos that use GitHub for issue tracking. 

I look forward to your opinions and more ideas to improve the current issue tracking and searching! 🚀 🤖 

### Guidelines

- [x] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by daguitosama** [comment]
yeah, i just  landed here looking for "AI for search GitHub issues", defenetily a very usefull feature to help find issues by semantics and meaning instead keyword matching

**Comment by nutspiano** [comment]
This might not have been feasible in June 2023 or October 2024, but it is now highly possible. Having AI-powered search in Issues would greatly help in both finding solutions to ongoing issues and reducing duplicates.

Extensions to the original idea:
* **AI search during issue creation:** When submitting a new issue, pause and prompt the user if similar issues are found, reducing duplicates. If no duplicates are detected, proceed as usual without interrupting the existing flow.
* **Convert to comment:** If the new issue overlaps with an existing one, suggest adding it as a comment to that issue instead. Provide an editable, AI-generated comment that the user can submit with one click (tagged as AI-generated if left unedited).
* **Link related issues:** Suggest linking the new issue to relevant existing ones. Either automate this or provide a simple interface for users to search and tag related issues.

**Reply by Finesse** [reply]
Looking for similar issues should be feasible for GPT models. Under the hood they convert the meaning of the text into vectors. Rephrasing text with the same meaning produces very close vectors (this is what T in GPT is about). So searching for similar issues should be as simple (or hard) as searching for the closest vectors.

As GitHub I user, I eagerly want a search that is capable of searching synonyms, and I want an issue deduplication tool.
